pennylane/math/utils.py

```diff
@@ -391,7 +451,7 @@ def requires_grad(tensor, interface=None):
     if interface == "torch":
         return getattr(tensor, "requires_grad", False)
 
-    if interface == "numpy":
+    if interface in {"numpy", "scipy"}:

```

Comment:
How come these were changed? Do the test cases cover the `"scipy"` path?

Reply:
Before when calling `_multi_dispatch` on a scipy sparse array it returned "numpy", but using `get_interface` it returned "scipy".

When merging both functions I decided to return "scipy", and to pass the tests I needed to add this here.

These lines are covered in the following tests:
https://github.com/PennyLaneAI/pennylane/blob/b96ac595e47cdc837110eb0f1ebe78447d2678a4/pennylane/devices/tests/test_measurements.py#L37